### PR TITLE
[Workers] Add Strict Compression, Strict Crypto compatibility flags

### PR DESCRIPTION
**Pull request by fhanau**
Add the new `strictCompression`compatibility flag, implemented in https://github.com/cloudflare/workerd/commit/4460d51f5fa9c5b4d73833508d3b91f4cd1b5eb4#diff-c7fa8ca455dd055174097625258de733986b4df090ed5e56b260ff2ed9b71b9a

It might be useful to get https://github.com/cloudflare/cloudflare-docs/pull/9244 in first as the flag added in that PR was added first and a comment refers to it. I am working on some more compat flag changes, but wanted to get this one in first.

### CODE REVIEW

**Comment by fhanau** [comment]
Added another commit for the [soon-to-be-merged](https://github.com/cloudflare/workerd/pull/734/files#diff-c7fa8ca455dd055174097625258de733986b4df090ed5e56b260ff2ed9b71b9a) strict crypto flag.

File: content/workers/_partials/_platform-compatibility-dates/strict-compression.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+name: "Strict compression error checking"
+# sort date incremented by 1, noCfBotManagementDefault has the same
+# compatibility date but was added first.
+sort_date: "2023-08-02"
+enable_date: "2023-08-01"
+enable_flag: "strict_compression_checks"
+disable_flag: "no_strict_compression_checks"
+---
+
+Perform additional error checking in the Web Compression API and throw an error if a `DecompressionStream` has trailing data or gets closed before the full compressed data has been provided.
```
Comment (jasnell) [inline content/workers/_partials/_platform-compatibility-dates/strict-compression.md]: Nit: I need to do this also for some of the other compat flags, but including a code example illustrating the difference would be helpful for these. Non-blocking tho.

File: content/workers/_partials/_platform-compatibility-dates/strict-crypto.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+name: "Strict crypto error checking"
+# sort date increased by 2, noCfBotManagementDefault and strictCompression have
+# the same compatibility date but was added earlier.
+sort_date: "2023-08-03"
+enable_date: "2023-08-01"
+enable_flag: "strict_crypto_checks"
+disable_flag: "no_strict_crypto_checks"
+---
+
+Perform additional error checking in the Web Crypto API to conform with the specification. Additionally, reject key parameters that may be unsafe based on the modulus or key length.
```
Comment (jasnell) [inline content/workers/_partials/_platform-compatibility-dates/strict-crypto.md]: Examples of the kinds of additional error checks would be good to show here.

**Review by irvinebroque — APPROVED** [review]
+1 code examples ideal but 🚀

**Comment by irvinebroque** [comment]
Linking to this from https://github.com/cloudflare/cloudflare-docs/pull/9312/files#diff-8619da0887af5a02cc96d0335c7a4583d284b2d0dc59f19def45f58c58ec9e92R16 would be ideal

File: content/workers/_partials/_platform-compatibility-dates/strict-compression.md (hunk 1 above)
Comment (fhanau) [reply content/workers/_partials/_platform-compatibility-dates/strict-compression.md]: Would that be examples of code that would be rejected/how to fix it? For the crypto errors, I can describe the errors in more detail by describing what bounds are placed on the key/modulus. For the compression errors, needing to finish reading the data before closing the stream as well as throwing when there is trailing data are well-defined concepts, so I'm not sure how a good, concise example would look like.

**Pull request merged**